model-optimizer/extensions/ops/sparse_reshape.py

```diff
@@ -28,26 +29,77 @@ def supported_attrs(self):
 
     @staticmethod
     def infer(node: Node):
+        name = node.soft_get('name', node.id)
+
         input_indices_shape = node.in_port(0).data.get_shape()
         input_indices_value = node.in_port(0).data.get_value()
-        input_shape_value = node.in_port(1).data.get_value()
-        new_shape_value = node.in_port(2).data.get_value()
+        input_shape = node.in_port(1).data.get_value()
+        new_shape = node.in_port(2).data.get_value()
         new_shape_shape = node.in_port(2).data.get_shape()
 
-        assert input_shape_value is not None and new_shape_value is not None, \
+        assert input_shape is not None and new_shape is not None, \
             "Values for input shape and new shape must be defined"
-        assert np.count_nonzero(new_shape_value == -1) <= 1, \
+        assert np.count_nonzero(new_shape == -1) <= 1, \
             "Value -1 occurs in new shape value more than once"
 
-        node.out_port(1).data.set_shape(new_shape_shape)
-        output_shape_value = new_shape_value
-        if np.count_nonzero(output_shape_value == -1) == 1:
-            elem = np.prod(input_shape_value) // np.prod(new_shape_value[new_shape_value != -1])
-            output_shape_value[output_shape_value == -1] = elem
-        node.out_port(1).data.set_value(output_shape_value)
+        num_of_input_elements = np.prod(input_shape)
```

Comment:
what if `input_shape` contains dynamic_value?

Reply:
if `input_shape` contains dynamic values when these values will not be taken into account during product, e.g. `np.prod([dyn, 30]) = 30`. 

Regardless of this if input_shape is not fully defined `num_of_input_elements` will not be used anywhere